atom/browser/api/atom_api_system_preferences_mac.mm

```diff
@@ -8,13 +8,57 @@
 
 #import <Cocoa/Cocoa.h>
 
+#include "atom/browser/mac/atom_application.h"
 #include "atom/browser/mac/dict_util.h"
 #include "atom/common/native_mate_converters/gurl_converter.h"
 #include "atom/common/native_mate_converters/value_converter.h"
 #include "base/strings/sys_string_conversions.h"
 #include "base/values.h"
+#include "native_mate/object_template_builder.h"
 #include "net/base/mac/url_conversions.h"
 
+namespace mate {
+template <>
+struct Converter<NSAppearance*> {
+  static bool FromV8(v8::Isolate* isolate,
+                     v8::Local<v8::Value> val,
+                     NSAppearance** out) {
+    if (val->IsNull()) {
+      *out = nil;
+    } else {
+      std::string name;
+      if (!mate::ConvertFromV8(isolate, val, &name)) {
+        return false;
+      }
+      if (name == "light") {
+        *out = [NSAppearance appearanceNamed:NSAppearanceNameAqua];
+        return true;
+      } else if (name == "dark") {
+        *out = [NSAppearance appearanceNamed:NSAppearanceNameDarkAqua];
+        return true;
+      }
```

Comment:
Should there be a warning here when the name is neither 'light' nor 'dark'?

Reply:
The return false in the converter will cause an automatic error to be thrown because the provided value can't be conveterted into an appaearance